### DISCUSSION TITLE
Delays with image preloading in Safari

### DISCUSSION THREAD

**Discussion by sankalp35** [jspsych/jsPsych#1605] [Q&A]
Hello,

I am creating a task in which participants are shown a series of faces on screen (using the image-keyboard-response plugin). The faces are pulled in through a timeline variable, and the face array is inputted in the preload image section of jsPsych.init. All of this works well in Chrome, and the images are preloaded perfectly, but in Safari there is a clear lag/delay which means I can't rely on data from participants using Safari. Any leads on why this is, and how to fix it?

Sankalp

**Comment by jodeleeuw** [comment]
Hi Sankalp,

What version of jsPsych are you using?

**Reply by sankalp35** [reply]
Hi Josh,

It was probably jsPsych 6 before (or that's the folder name at least). I updated it to 6.3.0 and it still has the same problem. If it helps at all, the files are served through JATOS. 

Best
Sankalp

**Comment by becky-gilbert** [comment]
Hi @sankalp35!
Could this be the same issue as this one: #1548? 
If you open the [network tab in Safari's developer tools](https://developer.apple.com/safari/tools/) and then refresh the page, do you see all of the image files load when the experiment starts? And do they load again at the start of each trial?

**Reply by sankalp35** [reply]
Hi @becky-gilbert ,

Yes, it does look like a similar issue, if not the same as #1548. I'll look into that later, but to answer your question quickly, yes the images all load at the beginning, and then reload at the start of the trial (I can see the downloaded file size go up every trial in the network tab). Also something to note is that I have reverted back to jsPsych 6, due to a conditional function problem raised here #1611. Not sure if it is relevant to this problem, though!

**Reply by sankalp35** [reply]
Hey @becky-gilbert 
I have a quick update on this. I read some of the threads you've referred to above, and setting render_on_canvas: false does not help with Safari. There is some reason why safari is loading the preloaded content again.
